File: leetcode/code/389.find_the_diffrence.py
class Solution(object):
    def findTheDiffrence(self, s, t):
        temp_str = "".join([s, t])
        result = ord(temp_str[0])

        for c in temp_str[1:]:
            result ^= ord(c)

        return chr(result)


if __name__ == "__main__":
    s = Solution()
    assert s.findTheDiffrence("abcd", "abcde") == "e"

File: leetcode/code/389.find_the_diffrence.go
package main

import "fmt"

func findTheDiffrence(s, t string) rune {
	cache := make(map[rune]bool)

	for _, r := range s {
		cache[r] = true
	}

	var result rune
	for _, r := range t {
		if _, ok := cache[r]; !ok {
			result = r
			break
		}
	}

	return result
}

func main() {
	fmt.Printf("%c\n", findTheDiffrence("abcd", "abcde"))
}
